tar/siginfo.c

```diff
@@ -222,3 +225,24 @@ siginfo_done(struct bsdtar *bsdtar)
 	/* Free internal data structure. */
 	free(bsdtar->siginfo);
 }
+
+/**
+ * siginfo_raisetotal(total):
+ * Look at how many bytes on disk have been processed since the last update,
+ * and trigger a siginfo_printinfo() if it exceeds tarsnap_opt_processbytes.
+ */
+static void
+siginfo_raisetotal(int64_t total)
```

Comment:
Why not take the `bsdtar` structure here (or possibly `bsdtar->siginfo`)?  And put `lastprogress` into said structure?